httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseCachingPolicy.java

```diff
@@ -320,4 +360,100 @@ private boolean from1_0Origin(final HttpResponse response) {
+     * This method provides a basic implementation of the freshness lifetime calculation algorithm and may not be suitable
+     * for all use cases. Developers should consult the HTTP caching specifications for more information and consider
+     * implementing additional caching mechanisms as needed.
+     * </p>
+     *
+     * @param response the HTTP response for which to calculate the freshness lifetime
+     * @return the freshness lifetime of the response, in seconds
+     */
+    private Duration calculateFreshnessLifetime(final HttpResponse response) {
+        // Check if s-maxage is present and use its value if it is
+        final Header cacheControl = response.getFirstHeader(HttpHeaders.CACHE_CONTROL);
+        if (cacheControl == null) {
+            // If no cache-control header is present, assume no caching directives and return a default value
+            return DEFAULT_FRESHNESS_DURATION; // 5 minutes
+        }
+
+        final String cacheControlValue = cacheControl.getValue();
+        final String[] headerValues = cacheControlValue.split(",");
+
+        long maxAge = -1;
+        long sharedMaxAge = -1;
+
+        for (final String headerValue : headerValues) {
+            final Tokenizer.Cursor cursor = new Tokenizer.Cursor(0, headerValue.length());
+            while (!cursor.atEnd()) {
+                final String token = tokenParser.parseToken(headerValue, cursor, DELIMITER);
+                if (token.trim().startsWith("max-age=")) {
+                    try {
+                        maxAge = Long.parseLong(token.trim().substring("max-age=".length()));
+                    } catch (final NumberFormatException e) {
+                        // Handle malformed header value
+                        maxAge = -1;
+                        if (LOG.isWarnEnabled()) {
+                            LOG.warn("Malformed 'max-age' header value '{}'", token);
+                        }
+                    }
+                } else if (token.trim().startsWith("s-maxage=")) {
+                    try {
+                        sharedMaxAge = Long.parseLong(token.trim().substring("s-maxage=".length()));
+                    } catch (final NumberFormatException e) {
+                        // Handle malformed header value
+                        sharedMaxAge = -1;
+                        if (LOG.isWarnEnabled()) {
+                            LOG.warn("Malformed 's-maxage' header value '{}'", token);
+                        }
+                    }
+                }
+            }
+        }
+
+        if (sharedMaxAge != -1) {
+            return Duration.ofSeconds(sharedMaxAge);
+        } else if (maxAge != -1) {
+            return Duration.ofSeconds(maxAge);
+        }
+
+        // Check if Expires is present and use its value minus the value of the Date header
+        Instant expiresInstant = null;
+        Instant dateInstant = null;
+        if (response.containsHeader(HttpHeaders.EXPIRES)) {
```

Comment:
@arturobernalg Please do not do that. Please do not use two method calls where a single method call and a null check should perfectly suffice.